engine/common/follower/compliance_core.go

```diff
@@ -141,9 +143,9 @@ func (c *ComplianceCore) OnBlockRange(originID flow.Identifier, batch []*flow.Bl
 		// 2. The QC within the block is valid. A valid QC proves validity of all ancestors.
 		err := c.validator.ValidateProposal(hotstuffProposal)
 		if err != nil {
-			if model.IsInvalidBlockError(err) {
-				// TODO potential slashing
-				log.Err(err).Msgf("received invalid block proposal (potential slashing evidence)")
+			if invalidBlockError, ok := model.AsInvalidBlockError(err); ok {
+				// TODO: potential slashing
```

Comment:
I feel we can remove this TODO. The logic here is fulfilling its part of the contract by emitting a notification. 

```suggestion
```

Reply:
https://github.com/onflow/flow-go/pull/4174/commits/2afe79e0d7360acffc30d4bfb1301cbba7a86e44